### PR TITLE
fix:removed the callback from the logs where ever it is not necessary

### PR DESCRIPTION
**Pull request by vishal-kanna**
## Description

As per the issue , i have removed the unnecessary usage of the callback in logs  of msg_server.

closes: #3404 

### Commit Message / Changelog Entry

```bash
nit: remove "callback" from msg server logging, except where relevant.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/keeper/msg_server.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func (k Keeper) ChannelOpenTry(goCtx context.Context, msg *channeltypes.MsgChann
 		portCap, msg.Channel.Counterparty, msg.CounterpartyVersion, msg.ProofInit, msg.ProofHeight,
 	)
 	if err != nil {
-		ctx.Logger().Error("channel open try callback failed", "error", errorsmod.Wrap(err, "channel handshake open try failed"))
+		ctx.Logger().Error("channel open try  failed", "error", errorsmod.Wrap(err, "channel handshake open try failed"))
```
Comment (colin-axner) [inline modules/core/keeper/msg_server.go:236]:
```suggestion
		ctx.Logger().Error("channel open try failed", "error", errorsmod.Wrap(err, "channel handshake open try failed"))
```

File: modules/core/keeper/msg_server.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func (k Keeper) ChannelOpenTry(goCtx context.Context, msg *channeltypes.MsgChann
 	// Lookup module by port capability
 	module, portCap, err := k.PortKeeper.LookupModuleByPort(ctx, msg.PortId)
 	if err != nil {
-		ctx.Logger().Error("channel open try callback failed", "port-id", msg.PortId, "error", errorsmod.Wrap(err, "could not retrieve module from port-id"))
+		ctx.Logger().Error("channel open try  failed", "port-id", msg.PortId, "error", errorsmod.Wrap(err, "could not retrieve module from port-id"))
```
Comment (colin-axner) [inline modules/core/keeper/msg_server.go:220]:
```suggestion
		ctx.Logger().Error("channel open try failed", "port-id", msg.PortId, "error", errorsmod.Wrap(err, "could not retrieve module from port-id"))
```

**Review by colin-axner — COMMENTED** [review]
Thanks @vishal-kanna! The `callback` name can be removed from the `GetRoute()` failure calls as well. In addition, it seems some of the modified logging have additional spaces, see my suggestions below

**Review by crodriguezvega — APPROVED** [review]
Thanks, @vishal-kanna. I addressed the comments that @colin-axner mentioned. It should be good now.

**Comment by vishal-kanna** [comment]
Thanks @crodriguezvega for adding removing the comments.

**Comment by DimitrisJim** [comment]
Thanks! Grepping through the pr locally I'm seeing three additional cases of `callback` being included when they probably shouldn't.

In `RecvPacket`:

https://github.com/cosmos/ibc-go/blob/b562f24a1dbda9ea0bf7eae0890a9868abd6c0a7/modules/core/keeper/msg_server.go#L483

In `Timeout`:

https://github.com/cosmos/ibc-go/blob/b562f24a1dbda9ea0bf7eae0890a9868abd6c0a7/modules/core/keeper/msg_server.go#L554

and in `TimeoutOnClose`:

https://github.com/cosmos/ibc-go/blob/b562f24a1dbda9ea0bf7eae0890a9868abd6c0a7/modules/core/keeper/msg_server.go#L628

**Review by colin-axner — APPROVED** [review]
LGTM, thanks @vishal-kanna :)

**Pull request closed without merging**